pkg/asset/installconfig/aws/validation.go

```diff
@@ -166,6 +184,167 @@ func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorLi
 	return field.ErrorList{field.Required(field.NewPath("platform", "aws", "amiID"), "AMI must be provided")}
 }
 
+// eipAllocationExists checks if a given EIP allocation ID exists.
+func eipAllocationExists(svc ec2iface.EC2API, allocationID string) (bool, error) {
+	input := &ec2.DescribeAddressesInput{
+		AllocationIds: aws.StringSlice([]string{allocationID}),
+	}
+
+	_, err := svc.DescribeAddresses(input)
+	if err != nil {
+		// Handle specific AWS error for non-existent allocation ID
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "InvalidAllocationID.NotFound" {
+			return false, fmt.Errorf("The allocation ID does not exist")
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+// validateEIPAllocations validates an array of EIP allocation IDs based on given rules.
+func validateEIPAllocationsList(allocationIDs []string) error {
+	// Rule 1: Check for duplicates
+	seen := make(map[string]bool)
+	for _, id := range allocationIDs {
+		if seen[id] {
+			return fmt.Errorf("duplicate allocation ID found: %s", id)
```

Comment:
*nit* `allocation ID` is quite generic, doesn't mention that's an EIP, so it might make it clearer for users. Also, this error message should probably say it's not allowed vs. it was found.

```suggestion
			return fmt.Errorf("cannot have duplicate EIP Allocation IDs: %s", id)
```

Mind fixing the other occurrences too?

Reply:
done